src/app.ts

```diff
@@ -13,14 +13,38 @@ interface TemperatureSummary {
   average: number
 }
 
+const temperatures: Array<TemperatureReading> = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  temperatures.push(...readings);
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
-}
+  const filteredTemperatures = temperatures.filter(value => value.time.getTime() === new Date(date).getTime() && value.city === city);
+  const totalFilteredTemperatures = filteredTemperatures.length;
+
+  if (totalFilteredTemperatures > 0) {
+    const summary: TemperatureSummary = {
+      first: filteredTemperatures[0].temperature,
+      last: filteredTemperatures[totalFilteredTemperatures - 1].temperature,
+      high: filteredTemperatures[0].temperature,
+      low: filteredTemperatures[0].temperature,
+      average: filteredTemperatures[0].temperature,
+    };
+
+    for (let i = 1; i < totalFilteredTemperatures; i++) {
+      summary.high = filteredTemperatures[i].temperature > summary.high ? filteredTemperatures[i].temperature : summary.high;
+      summary.low = filteredTemperatures[i].temperature < summary.low ? filteredTemperatures[i].temperature : summary.low;
+      summary.average += filteredTemperatures[i].temperature;
```

Comment:
You shouldn't use the same variable for accumulator and for average at the same time. You can use a different variable with a meaning name like `acum`. And then calculate the average using that variable.